pkgs/development/tools/revive/default.nix

```
@@ -0,0 +1,52 @@
+{ buildGoModule, fetchFromGitHub, lib }:
+
+buildGoModule rec {
+  pname = "revive";
+  version = "1.2.1";
+
+  src = fetchFromGitHub {
+    owner = "mgechev";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-XBdid/Ze9g385dMup2Zo3RicL9k6NMjdBG6RMdsJn24=";
+    # populate values that require us to use git. By doing this in postFetch we
+    # can delete .git afterwards and maintain better reproducibility of the src.
+    leaveDotGit = true;
+    postFetch = ''
+      git -C "$out" rev-parse HEAD > $out/COMMIT
+      rm -rf "$out/.git"
+    '';
+  };
+  vendorSha256 = "sha256-Fpl5i+qMvJ/CDh8X0gps9C/BxF7/Uvln+3DpVOXE0WQ=";
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X github.com/mgechev/revive/cli.version=${version}"
+    "-X github.com/mgechev/revive/cli.builtBy=nix"
+  ];
+
+  # ldflags based on metadata from git and source
+  preBuild = ''
+    ldflags+=" -X github.com/mgechev/revive/cli.commit=$(cat COMMIT)"
+    ldflags+=" -X github.com/mgechev/revive/cli.date=$(date -u +%Y-%m-%dT%H:%MZ)"
```

addressing https://github.com/NixOS/nixpkgs/pull/176812#discussion_r907858818

```suggestion
    ldflags+=" -X 'github.com/mgechev/revive/cli.date=$(cat SOURCE_DATE_EPOCH)'"
```

should solve the spaces issues, the single quotes in double quotes wont cause issues for the `$()`

```
λ echo " -X 'github.com/mgechev/revive/cli.date=$(date -u '+%Y-%m-%d %H:%M UTC')'"
 -X 'github.com/mgechev/revive/cli.date=2022-06-29 13:13 UTC'
```

(remember to update the sha)

---

Done! Thank you!